scipy/stats/_stats_py.py

```diff
@@ -4658,7 +4658,7 @@ def statistic(x, y, axis):
         warnings.warn(stats.NearConstantInputWarning(msg), stacklevel=2)
 
     with np.errstate(invalid='ignore', divide='ignore'):
-        r = xp.sum(xm/normxm * ym/normym, axis=axis)
+        r = xp.linalg.vecdot(xm / normxm, ym / normym, axis=axis)
```

Comment:
Do we need to worry about checking if the `linalg` extension is implemented? https://data-apis.org/array-api/latest/extensions/linear_algebra_functions.html#extension-name-and-usage

Reply:
I suppose I assumed that `array_api_compat` already would. If not, and If we don't already have an `xp_vecdot` that does that, sounds like something for `_array_api.py` or `array-api-extra`.

Reply:
But it might also be valid to error out of it's not implemented. If we're not going to provide a generic implementationsl, I don't see a big difference between checking and raising vs just raising when it fails.

Reply:
Oops, I should have just called `vecdot`, not `linalg.vecdot`.

Reply:
Sounds good, the rest looks good too